cadquery/occ_impl/shapes.py

```diff
@@ -5432,46 +5541,142 @@ def _bool_op(
     builder.Build()
 
 
-def fuse(s1: Shape, s2: Shape, tol: float = 0.0) -> Shape:
+def _set_glue(builder: BOPAlgo_Builder, glue: GlueLiteral):
+
+    if glue:
+        builder.SetGlue(
+            BOPAlgo_GlueEnum.BOPAlgo_GlueFull
+            if glue == "full"
+            else BOPAlgo_GlueEnum.BOPAlgo_GlueShift
+        )
+
+
+def _set_builder_options(builder: BOPAlgo_Builder, tol: float):
+
+    builder.SetRunParallel(True)
+    builder.SetUseOBB(True)
+    builder.SetNonDestructive(True)
+
+    if tol:
+        builder.SetFuzzyValue(tol)
+
+
+def setThreads(n: int):
     """
-    Fuse two shapes.
+    Set number of threads to be used by boolean operations.
     """
 
-    builder = BRepAlgoAPI_Fuse()
-    _bool_op(s1, s2, builder, tol)
+    pool = OSD_ThreadPool.DefaultPool_s()
```

Comment:
Could this have implications for GUIs and client-server systems like CQ-editor, jupyter-cadquery, etc? Multi-threading seems to get weird sometimes in those types of applications.

Reply:
I guess if it does cause an issue, the user can just not use this feature.

Reply:
You indeed, don't need to use this future. I doubt that this has implications for GUIs, as OCCT (and CQ) already uses multi-threading and this simply allows to change the default threadpool size.